### PR TITLE
Clippy needless_borrow and single_match warnings fixed

### PR DESCRIPTION
**Pull request by ggaspersic**
Resolving the following clippy warnings:
- https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrow
- https://rust-lang.github.io/rust-clippy/master/index.html#single_match

Smaller additional changes:
- output buffer truncated before filled in parser.rs to prevent unnecessary slice creation
- inlined if into else in block_ffm.rs
- simplified gz var cache.rs

### CODE REVIEW

File: src/block_ffm.rs [hunk 1]
```diff
@@ -857,17 +857,15 @@ impl<L: OptimizerTrait + 'static> BlockFFM<L> {
                         ffm_weights.get_unchecked(feature_index + z) * feature_value;
                 }
             }
+        } else if feature_value == 1.0 {
+            for z in 0..field_embedding_len {
```
Comment (SkBlaz) [inline src/block_ffm.rs:861]: Does this compile to something different actually?

File: src/persistence.rs [hunk 2]
```diff
@@ -113,11 +113,8 @@ fn load_regressor_without_weights(
     let mut mi = model_instance::ModelInstance::new_from_buf(input_bufreader)
         .expect("Loading model instance from regressor failed");
 
-    match cmd_arguments {
-        Some(cmd_args) => {
-            model_instance::ModelInstance::update_hyperparameters_from_cmd(cmd_args, &mut mi)?;
-        }
-        _ => (),
+    if let Some(cmd_args) = cmd_arguments {
+        model_instance::ModelInstance::update_hyperparameters_from_cmd(cmd_args, &mut mi)?;
```
Comment (SkBlaz) [inline src/persistence.rs:117]: I think we discussed let Some vs. match sometime ago - not sure what was the preferred choice @bbenshalom wdyt?

Comment (bbenshalom) [reply src/persistence.rs:117]: I'm fine with either - "if let" doesn't antagonize clippy so we can keep it

Comment (ggaspersic) [reply src/persistence.rs:117]: This was actually a suggestion by clippy :)

Comment (SkBlaz) [reply src/persistence.rs:117]: Yes, well aware :)

**Review by SkBlaz — APPROVED** [review]

**Pull request merged**